cmd/oras/internal/output/print.go

```diff
@@ -57,7 +57,7 @@ func (p *Printer) Println(a ...any) error {
 	_, err := fmt.Fprintln(p.out, a...)
 	if err != nil {
 		err = fmt.Errorf("display output error: %w", err)
-		_, _ = fmt.Fprint(os.Stderr, err)
+		_, _ = fmt.Fprint(p.err, err)
```

Comment:
What if a user redirects stderr into a malformed output? In that case they won't be able to see the error message anyway.

IMO here we should always output to os.Stderr to make sure the `err` is delivered in a best-effort way.

Reply:
If we don't pass `cmd.ErrOrStderr()` to the printer, the user cannot capture it for go scripting. That is the point of this.

The user can always mess things up, redirect stderr to /dev/null or whatever they want to do.

Reply:
Okay there are two kinds of redirection. Well maybe I am not making it clear, by redirection I mean: during go scripting, changing the command stderr via https://pkg.go.dev/github.com/spf13/cobra@v1.8.1#Command.SetErr.

Reply:
In that case should we sent `err` here to os.Stderr or the user specified error writer? The former makes more sense to me.

Reply:
The key point here is: if a user runs a binary that uses ORAS go scripting, should the `err` here be consider as:

1) ORAS' error output? (Should sent it to `cmd.ErrOrStderr()`), or
2) the binary's error output?  (Should sent it to `os.Stdout`)

Reply:
I am okay with sending stdout output failure to user-specified command stderr, but that may change in the future. Is there any way we can declare that ORAS go scripting(using ORAS CLI as golang SDK) is experimental and subject to changes?

Reply:
We can follow it up in https://github.com/oras-project/oras/issues/1432 and https://github.com/oras-project/oras-www/issues/357